src/pattern.go

```diff
@@ -34,6 +35,11 @@ type term struct {
 	caseSensitive bool
 }
 
+// String returns the string representation of a term.
+func (t term) String() string {
+	return fmt.Sprintf("term{typ: %d, inv: %v, text: []rune(%q), caseSensitive: %v}", t.typ, t.inv, string(t.text), t.caseSensitive)
+}
```

Comment:
String representations are kind of an art. I tried to follow the Go syntax representation here. Please let me know if you prefer another format. I'm happy to change it.